common/lib/xmodule/xmodule/conditional_module.py

```diff
@@ -265,7 +267,7 @@ def __init__(self, *args, **kwargs):
         if not self.sources_list:
             if 'sources' in self.xml_attributes and isinstance(self.xml_attributes['sources'], basestring):
                 self.sources_list = [
-                    self.location.course_key.make_usage_key_from_deprecated_string(item)
+                    UsageKey.from_string(item).map_into_course(self.location.course_key)
```

Comment:
@nasthagiri This change makes a [test](https://github.com/edx/edx-platform/blob/e8a36957b1f732974260e7b9b42b9c25148b492c/cms/djangoapps/contentstore/tests/test_import.py#L220) fail because the two bits of code are not actually equivalent.

`self.location.course_key.make_usage_key_from_deprecated_string(item)` actually just [substitutes a run](https://github.com/edx/opaque-keys/blob/ffb2ee1d1be8bb051c32a538067743f74d8503f4/opaque_keys/edx/locator.py#L297) into the course_key, and for old mongo, that is a no-op. Therefore the conditional source maintains the course ID that is [specified in the XML file](https://github.com/edx/edx-platform/blob/6d2468d0dcf782a7561ecc61946146d6fae2762f/common/test/data/conditional/conditional/condone.xml#L1).

However, `UsageKey.from_string(item).map_into_course(self.location.course_key)` will actually [replace the original course ID](https://github.com/edx/opaque-keys/blob/ffb2ee1d1be8bb051c32a538067743f74d8503f4/opaque_keys/__init__.py#L329) with the new course ID.

Given the existence of this test, it appears this is not the desired functionality (though [note the comment](https://github.com/edx/edx-platform/blob/e8a36957b1f732974260e7b9b42b9c25148b492c/cms/djangoapps/contentstore/tests/test_import.py#L201) at the top of the test that acknowledges different behavior in split vs. old mongo).

I will try to touch base with you about what we should do.